ebuild-writing/misc-files/patches/text.xml

```diff
@@ -331,6 +331,34 @@ scrub_patch some-patch-you-found.patch
 </body>
 </subsection>
 
+<subsection>
+<title>Conditional Patching</title>
+<body>
+<p>
+Patching is ideally only done to make the package in question build properly,
+and should not be done to modify the runtime behaviour of the package. This is
+what USE flags and features of the package are for. As such, it is preferable to
+apply patches unconditionally and avoid conditional patching.
+</p>
+<p>
+Consider the following example <c>src_prepare()</c> implementation:
+</p>
+<codesample lang="ebuild">
+src_prepare() {
+	if use threads; then
+		PATCHES+=( "${FILESDIR}/${P}"-mt.patch )
+	fi
+	default
+}
+</codesample>
+<p>
+As this patch is only applied when <c>USE="threads"</c> is set, any developer
+creating new versions of this package might not detect whether the patch applies
+successfully if they don't test with the same flag.
```

Comment:
Let's add in a new paragraph that this can become unsustainable too, or even impossible to test, with USE flag combinatorics. So, while it's undesirable for even one or two patches, it becomes truly untestable and unfit for production with several flags if more than one applies patches.

We should also explicitly state that patches should be written such that they affect behaviour correctly based on e.g. build time definitions or config options rather than USE flags directly, allowing them to be applied unconditionally.

(Also, for e.g. musl patches, we want a portable fix, not a hack which is only applied _for_ musl, which kind of makes the problem worse.)

Nice work!